tools/ot/src/main/java/dk/alexandra/fresco/tools/ot/base/NaorPinkasOt.java

```diff
@@ -0,0 +1,323 @@
+   * parameters.
+   *
+   * @param myId
+   *          The ID of the calling party
+   * @param otherId
+   *          The ID of the other party
+   * @param randBit
+   *          The calling party's secure randomness generator
+   * @param network
+   *          The underlying network to use
+   * @param params
+   *          The Diffie-Hellman parameters to use
+   */
+  public NaorPinkasOt(int myId, int otherId, Drbg randBit, Network network,
+      DHParameterSpec params) {
+    this.myId = myId;
+    this.otherId = otherId;
+    this.randBit = randBit;
+    this.network = network;
+    this.hashDigest = ExceptionConverter.safe(() -> MessageDigest.getInstance(
+        HASH_ALGORITHM),
+        "Missing secure, hash function which is dependent in this library");
+    this.params = params;
+    this.randNum = new DrngImpl(randBit);
+  }
+
+  public DHParameterSpec getDhParams() {
+    return params;
+  }
+
+  public void setDhParams(DHParameterSpec newParams) {
+    this.params = newParams;
+  }
+
+  @Override
+  public void send(StrictBitVector messageZero, StrictBitVector messageOne) {
+    // To avoid leakage both messages sent will have the length of the longest
+    // of the two
+    int maxBitLength = Math.max(messageZero.getSize(), messageOne.getSize());
+    Pair<byte[], byte[]> seedMessages = sendBytesOt();
+    // We divide the length with 8 to get the byte length and then use the short
+    // messages from the OT as seeds for a PRG which does a one-time-pad
+    // encryption of the true messages to send
+    byte[] encryptedZeroMessage = padMessage(messageZero.toByteArray(),
+        maxBitLength / 8, seedMessages.getFirst());
+    byte[] encryptedOneMessage = padMessage(messageOne.toByteArray(),
+        maxBitLength / 8, seedMessages.getSecond());
+    // Send the padded messages
+    network.send(otherId, encryptedZeroMessage);
+    network.send(otherId, encryptedOneMessage);
+  }
+
+  @Override
+  public StrictBitVector receive(Boolean choiceBit) {
+    byte[] seed = receiveByteOt(choiceBit);
+    byte[] encryptedZeroMessage = network.receive(otherId);
+    byte[] encryptedOneMessage = network.receive(otherId);
+    return recoverTrueMessage(encryptedZeroMessage, encryptedOneMessage, seed,
+        choiceBit);
+  }
+
+  /**
+   * Make a one-time padding of a {@code message} using a PRG seeded on
+   * {@code seed}.
+   *
+   * @param message
+   *          The message to one-time pad
+   *
+   * @param maxSize
+   *          The size of the resultant ciphertext. MUST be at least the length
+   *          of the message itself. If longer, it will be padded with zeros
+   * @param seed
+   *          The seed to use for a PRG
+   * @return The one-time padded message
+   */
+  private byte[] padMessage(byte[] message, int maxSize,
+      byte[] seed) {
+    byte[] maxLengthMessage = Arrays.copyOf(message, maxSize);
+    byte[] encryptedMessage = new byte[maxSize];
+    Drbg prg = new AesCtrDrbg(seed);
```

Comment:
As we discussed, this should not really be needed. It's fine to fix the message lenght to the digest length. (in fact that is exactly what we need for our application).

Reply:
Just to be clear, you want me to change the code to thrown an exception if the messages given as input are of the same length as the PRG seed?